src/hoc/asyncComponent.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { Component } from 'react';
+
+const asyncComponent = (importComponent) => {
+	return class extends Component {
+		state = {
+			component: null
+		}
+
+		componentDidMount() {
+			importComponent()
+				.then(cmp => {
+					this.setState({ component: cmp.default });
+				});
+		}
+
+		render() {
+			const C = this.state.component;
```

Comment:
Rather than naming this variable as `C`. We should provide a more specific naming convention for code clarity.